### PR TITLE
NOISSUE - Add Development Guide doc

### PR DESCRIPTION
**Pull request by drasko**
Signed-off-by: drasko <drasko.draskovic@gmail.com>

### CODE REVIEW

**Review by chombium — CHANGES_REQUESTED** [review]
well done @drasko. You have written the missing manual which will save us a ton of discussions in the mailing list and the Gitter channel.

I would add few warnings and one optional step as well.

As an optional step I would describe few changes how to add a volume to the ```things-db``` and ```users-db``` container, so that the data doesn't lost when this container is rebuilt and
executing again the db initialization of the [users](https://github.com/mainflux/mainflux/blob/master/users/postgres/init.go) and [thing](https://github.com/mainflux/mainflux/blob/master/things/postgres/init.go) services, so that the db state is the latest.

The thing with the migrations is that it would probably need a bit of refactoring so that they can be executed whenever that's needed. We should probably also replace the "CREATE TABLE" with "CREATE TABLE IF NOT EXISTS" (available from pgsql version 9.1). Do you think that this needs a ticket?

As a warning I would add that when the executing ```make dockers``` or ```make docker_things-db``` or ```make docker_users-db``` the data in the pgsql container will be lost.

File: docs/dev-guide.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+mkdir -p $GOPATH/src/github.com/mainflux
+cd $GOPATH/src/github.com/mainflux
+git clone https://github.com/mainflux/mainflux.git
+cd mainflux
+```
+
+## Building
+
+### Prerequisites
+Make sure that you have [Protocol Buffers](https://developers.google.com/protocol-buffers/) compiler (`protoc`) installed.
```
Comment (dborovcanin) [inline docs/dev-guide.md:15]: I think that `protoc` version matters and that we currently use `3.5.1`. Maybe it's a good idea to mention that here.

File: docs/dev-guide.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+```
+make docker_http
+```
+
+> N.B. Mainflux creates `FROM scratch` docker containers which as compact and small in size.
+
+### MQTT Microservice
+MQTT Microservice in Mainflux is special, as it is currently the only microservice written in NodeJS. It is not compiled,
+but node modules need to be download in order to start the service:
```
Comment (anovakovic01) [inline docs/dev-guide.md]: `downloaded`

File: docs/dev-guide.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+npm install
+```
+
+After that MQTT Adapter can be started from top directory (as it needs to find `*.proto` files) with:
+```
+node mqtt/mqtt.js
+```
+
+### Protobuf
+Aforemntioned  `make` (which is an alias for `make all` target) is calling `protoc` command prior to compiling individual microservices.
```
Comment (anovakovic01) [inline docs/dev-guide.md]: `Aforementioned`

File: docs/dev-guide.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+To do this by hand, execute:
+
+```
+protoc --go_out=plugins=grpc:. *.proto
```
Comment (anovakovic01) [inline docs/dev-guide.md:89]: Maybe add that it can be done also by calling `make proto`.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

File: docs/dev-guide.md (hunk 1 above)
Comment (drasko) [reply docs/dev-guide.md:15]: I would leave this out of the generic doc, as it is just the implementation detail.

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Review by chombium — APPROVED** [review]

**Pull request merged**